### DISCUSSION TITLE
How to plot predictions for specific series when using DeepVAREstimator model?

### DISCUSSION THREAD

**Discussion by vikasmalhotra08** [awslabs/gluonts#1713] [Q&A]
Hello,

I have started using the DeepVAREstimator model for modeling 1000 series at a time. I am interested in extracting predictions for one series from M series and plot the predictions along with true values when using `make_evaluation_predictions` for validation. I am not sure if I am doing this right as my first stab at this seems to indicate I am not picking the right series. 

Here is my code and its description:
- Data Points: 1003 series over 1 year timeframe with daily cadence. That gives me a shape of (100, 30, 1003) when trying to validate over 100 samples. 
- Model used: DeepVAREstimator
- **Series that I am interested in:** I moved the series to start of the train dataset so that it would be easier to parse out. 
- Train DS looks like this:
```
{'target': array([[ 81.89907 ,  82.164955,  89.71054 , ..., 292.18173 , 282.51968 ,
         275.4792  ],
        [ 58.308044,  58.308044,  58.308044, ..., 242.2358  , 234.98232 ,
         226.46036 ],
        [ 58.308044,  58.308044,  58.308044, ..., 242.23631 , 234.98158 ,
         226.45839 ],
        ...,
        [ 58.308044,  58.934883,  59.519157, ..., 222.96513 , 223.66765 ,
         224.34708 ],
        [ 58.308044,  58.868057,  59.53403 , ..., 234.14537 , 234.92073 ,
         235.52625 ],
        [ 58.308044,  58.923264,  59.520733, ..., 215.28896 , 216.03981 ,
         216.71579 ]], dtype=float32),
 'start': Timestamp('2020-08-09 00:00:00', freq='D'),
 'source': SourceContext(source='list_data', row=0)}
```

My code:
```
def fetch_data():
     # Scaling the inputs to see if this has an impact or not. 
    train_df_backup = train_df.loc[:, train_df.columns != field_name]
    metric_df = train_df[field_name]
    
    # fir transform
    # Scale the features
    scaler = MinMaxScaler(feature_range=(train_df[field_name].min(), train_df[field_name].max()))
    train_df_backup[train_df_backup.columns] = scaler.fit_transform(train_df_backup[train_df_backup.columns])
    
    # Combine the field_name and other series
    final_df = pd.concat([metric_df, train_df_backup], axis=1)
    
    # Split the data into multiple sets    
    X_train_values = final_df.iloc[: -single_prediction_length]
    X_test_values  = final_df.iloc[-single_prediction_length:]
    
    # transpose the data for feeding into gluonts    
    train_cal_features = X_train_values.values.T
    test_cal_features = X_test_values.values.T    
    
    # setup dates
    train_dates = [pd.Timestamp("2020-08-09", freq='D') for _ in range(len(X_train_values))]
    test_dates = [pd.Timestamp("2021-03-08", freq='D') for _ in range(len(X_test_values))]
    
    # Prepare the datset according to gluonTS format:
    train_ds = ListDataset(
        [
            {
                FieldName.TARGET: train_cal_features,
                FieldName.START: train_dates[0]
            }
        ], 
        freq="D",
        one_dim_target=False
    )
    
    test_ds = ListDataset(
        [
            {
                FieldName.TARGET: test_cal_features,
                FieldName.START: test_dates[0]
            }
        ], 
        freq="D",
        one_dim_target=False
    )
    
    target_dim = train_cal_features.shape[0]
    
    return train_ds, test_ds, target_dim

train_ds, test_ds, target_dim = fetch_data()

# Model code
from gluonts.model.deepvar import DeepVAREstimator
from gluonts.mx.trainer import Trainer
from gluonts.torch.modules.scaler import MeanScaler
from functools import partial

estimator = DeepVAREstimator(
        freq = 'D',
        prediction_length = 30,
        target_dim = target_dim,
        scaling=False,
        trainer=Trainer(           
            epochs=2,
            batch_size=16,
        ),
    )

predictor = estimator.train(train_ds)
```

Training for 2 epochs leads to following loss:
```
100%|██████████| 50/50 [01:12<00:00,  1.45s/it, epoch=1/2, avg_epoch_loss=1.78e+6]
100%|██████████| 50/50 [01:16<00:00,  1.53s/it, epoch=2/2, avg_epoch_loss=7.28e+4]
```

Model evaluation was done like this:
```
# evaluate your model (should be done on a test set!)
from gluonts.evaluation import MultivariateEvaluator
from gluonts.evaluation.backtest import backtest_metrics

agg_metrics, _ = backtest_metrics(
        test_dataset=test_ds,
        predictor=predictor,
        evaluator=MultivariateEvaluator(
            quantiles=(0.5, 0.7, 0.9)
        ))

print(agg_metrics['0_RMSE'])

# draw some samples from the predicted multivariate distribution
from gluonts.evaluation.backtest import make_evaluation_predictions
from gluonts.evaluation import Evaluator

forecast_it, ts_it = make_evaluation_predictions(
    test_ds, predictor=predictor, num_samples=100
)

from tqdm import tqdm

print("Obtaining time series conditioning values ...")
tss = list(tqdm(ts_it, total=len(test_ds)))

print("Obtaining time series predictions ...")
forecasts = list(tqdm(forecast_it, total=len(test_ds)))

# first entry of the time series list
ts_entry = tss[0]
ts_entry = ts_entry[0]

print(forecasts[0].samples.shape) # 100 samples for 30 days over 1003 series

# fetch the first series from forecasts as that should be the one that I am interested in:
first_dim_forecast = forecasts[0].copy_dim(0)

first_dim_forecast.plot()
```

![image](https://user-images.githubusercontent.com/10645844/135646666-79f813bb-344e-4f2f-8a77-560cd88e1c4b.png)

The plot seems off to me. Either I am picking the wrong series or my model is not able to learn the series. Can someone point what can be the problem?
